### PR TITLE
fix: handle input binding response correctly for http protocol app

### PR DESCRIPTION
**Pull request by zcong1993**
# Description

The [HTTP binding api](https://docs.dapr.io/reference/api/bindings_api/#http-response-body-optional) says user's binding  http response body can be used to directly bind input bindings with state stores or output bindings, but it does not seem to be implemented. This pr will fix this.

## Issue reference

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by zcong1993** [comment]
any update? I've updated the description.

File: pkg/runtime/runtime.go [hunk 1]
```diff
@@ -918,6 +918,17 @@ func (a *DaprRuntime) sendBindingEventToApp(bindingName string, data []byte, met
 
 		if resp.Message().Data != nil && len(resp.Message().Data.Value) > 0 {
 			appResponseBody = resp.Message().Data.Value
+			err := a.json.Unmarshal(resp.Message().Data.Value, &response)
+			if err == nil && response.Data != nil {
+				b, errB := a.json.Marshal(response.Data)
+				if errB == nil {
+					appResponseBody = b
```
Comment (daixiang0) [inline pkg/runtime/runtime.go:925]: Please add tests if possible.

Comment (zcong1993) [reply pkg/runtime/runtime.go:925]: Sorry, I can't find any test code of grpc for this feature.

**Comment by yaron2** [comment]
@zcong1993 we have actually removed this functionality from Dapr on purpose a while back. If you see it in the docs, then it should be removed. We would very much appreciate a PR to the docs to remove this part.

**Pull request closed without merging**